src/ansys/hps/client/authenticate.py

```diff
@@ -85,41 +120,52 @@ def authenticate(
         JSON-encoded content of a :class:`requests.Response` object.
     """
 
-    auth_postfix = f"auth/realms/{realm}"
-    if url.endswith(f"/{auth_postfix}") or url.endswith(f"/{auth_postfix}/"):
-        auth_url = url
-    else:
-        auth_url = urllib.parse.urljoin(url + "/", auth_postfix)
-    log.debug(f"Authenticating using {auth_url}")
+    auth_url = str(auth_url)
+    disco_dict = get_discovery_data(auth_url, timeout)
+    token_url = disco_dict["token_endpoint"]
 
     with requests.Session() as session:
-        session.verify = verify
-        session.headers.update(
-            {"content-type": "application/x-www-form-urlencoded"},
-        )
-
-        token_url = f"{auth_url}/protocol/openid-connect/token"
+        session.verify = False
+        session.headers = ({"content-type": "application/x-www-form-urlencoded"},)
 
+        # seemingly necessary all the time
         data = {
             "client_id": client_id,
-            "grant_type": grant_type,
             "scope": scope,
         }
-        if client_secret is not None:
+
+        # If someone specifically calls out a grant type, just use it directly.
+        if grant_type and not grant_type.isspace():
+            data["grant_type"] = grant_type
+
+        # Many grant types can have client secrets, client secret with other things just means it not a public endpoint
+        if client_secret is not None and not client_secret.isspace():
```

Comment:
why here and in the following we check for `isspace`? An empty string doesn't mean it is unset, I think we should just check for `is not None`.

Reply:
I'll try to make sure I filter out weird things in the users of pyHPS then...